packages/braid-design-system/src/lib/components/TooltipRenderer/TooltipRenderer.tsx

```diff
@@ -138,207 +106,107 @@ export const TooltipRenderer = ({
-      const handleFocusIn = (event: FocusEvent) => {
-        if (event.currentTarget !== triggerRef) {
-          setControlledVisible(false);
-        }
-      };
+    const setArrowPosition = () => {
+      const tooltipPosition = tooltipRef.current?.getBoundingClientRect();
+      const triggerPosition = triggerRef.current?.getBoundingClientRect();
 
-      document.addEventListener('focusin', handleFocusIn);
+      if (!tooltipPosition || !triggerPosition) {
+        return;
+      }
 
-      return () => {
-        document.removeEventListener('focusin', handleFocusIn);
-      };
-    }
-  }, [triggerRef, visible]);
+      setInferredPlacement(
+        tooltipPosition.top > triggerPosition.top ? 'bottom' : 'top',
+      );
 
-  assert(
-    useEffect(() => {
-      if (tooltipRef) {
-        assert(
-          tooltipRef.querySelectorAll('button, input, select, textarea, a')
-            .length === 0,
-          'For accessibility reasons, tooltips must not contain interactive elements',
-        );
-      }
-    }, [tooltipRef]) === undefined,
-  );
+      const triggerLeft = triggerPosition.left;
+      const tooltipLeftToTriggerLeft =
+        triggerLeft - tooltipPosition.left - edgeOffsetAsPx;
 
-  useEffect(() => {
-    if (!tooltipRef || !visible) {
-      return setOpacity(0);
-    }
+      setArrowLeftOffset(tooltipLeftToTriggerLeft + triggerPosition.width / 2);
+    };
 
-    const timeout = setTimeout(() => setOpacity(100), isMobile() ? 0 : 250);
+    const timeoutId = setTimeout(() => {
+      const frameId = requestAnimationFrame(setArrowPosition);
+      return () => cancelAnimationFrame(frameId);
+    });
 
-    return () => clearTimeout(timeout);
-  }, [tooltipRef, visible]);
+    return () => clearTimeout(timeoutId);
+  }, [open, isStatic, edgeOffsetAsPx]);
 
   return (
     <>
-      {children({
-        triggerProps: {
-          tabIndex: 0,
-          ref: setTriggerRef,
-          'aria-describedby': resolvedId,
-        },
-      })}
-
-      {triggerRef && (
-        <BraidPortal>
-          <div
-            id={resolvedId}
-            role="tooltip"
-            hidden={!visible ? true : undefined}
-            className={atoms({
-              reset: 'div',
-              zIndex: 'notification',
-              pointerEvents: 'none',
-              display: triggerRef && visible ? undefined : 'none',
-            })}
-            {...(visible
-              ? getTooltipProps({
-                  ref: setTooltipRef,
-                })
-              : null)}
-          >
-            <TooltipContent opacity={opacity} arrowProps={getArrowProps()}>
-              {tooltip}
-            </TooltipContent>
-          </div>
-        </BraidPortal>
-      )}
+      <Box
+        tabIndex={-1}
+        onClick={() => isMobile() && setOpen(!open)}
+        onMouseEnter={() => !isMobile() && setOpen(true)}
+        onFocus={() => !isMobile() && setOpen(true)}
+        onMouseLeave={() => !isMobile() && setOpen(false)}
```

Comment:
Rather than calling `isMobile` in the handlers, it would be better to conditionally bind the handlers.

The `isMobile` check is a user agent check too (not screen size), so it can be set once as a ref too, rather than call multiple times on every render.